Sources/interoperability-tests/InteroperabilityTestsExecutable.swift

```diff
@@ -60,4 +60,95 @@ struct InteroperabilityTestsExecutable: AsyncParsableCommand {
       }
     }
   }
+
+  struct RunTest: AsyncParsableCommand {
+    static var configuration = CommandConfiguration(
+      abstract: "Runs a gRPC interoperability test using a gRPC Swift client."
+    )
+
+    @Option(help: "The host the server is running on")
+    var host: String
+
+    @Option(help: "The port to connect to")
+    var port: Int
+
+    @Argument(help: "The name of the test to run")
+    var testName: String
+
+    func run() async throws {
+      guard let testCase = InteroperabilityTestCase(rawValue: self.testName) else {
+        throw InteroperabilityTestError.testNotFound(name: self.testName)
+      }
+
+      try await Self.runTest(
+        testCase: testCase,
+        host: self.host,
+        port: self.port
+      )
+    }
+
+    internal static func runTest(
+      testCase: InteroperabilityTestCase,
+      host: String,
+      port: Int
+    ) async throws {
+      let group = MultiThreadedEventLoopGroup(numberOfThreads: 1)
+      defer {
+        try! group.syncShutdownGracefully()
+      }
+
+      var transportConfig = HTTP2ClientTransport.Posix.Config.defaults
+      transportConfig.compression.enabledAlgorithms = .all
+      let serviceConfig = ServiceConfig(loadBalancingConfig: [.roundRobin])
+      let transport = try HTTP2ClientTransport.Posix(
+        target: .ipv4(host: "0.0.0.0", port: port),
+        config: transportConfig,
+        serviceConfig: serviceConfig,
+        eventLoopGroup: group
+      )
+      let client = GRPCClient(transport: transport)
+
+      try await withThrowingTaskGroup(of: Void.self) { group in
+        group.addTask {
+          try await client.run()
+        }
+
+        group.addTask {
+          print("Running '\(testCase.name)' ... ", terminator: "")
+          do {
+            try await testCase.makeTest().run(client: client)
+            print("PASSED")
+          } catch {
+            print("FAILED")
+            throw InteroperabilityTestError.testFailed(cause: error)
```

Comment:
If we run more than one test we shouldn't throw an error at this point, we should wait for all tests to run first.